pkg/utils/nodeutils.go

```diff
@@ -213,6 +214,37 @@ func GetKCFromMC(mc *mcfgv1.MachineConfig, client runtimeclient.Client) (*mcfgv1
 	return nil, fmt.Errorf("machine config %s doesn't have a KubeletConfig owner reference", mc.GetName())
 }
 
+func getEncodedKC(value interface{}) (interface{}, error) {
+	if value == nil {
+		return kcPath, fmt.Errorf("value is nil")
+	}
+
+	index := 0
+	for {
+		fileContent, err := jsonpath.Get("storage.files["+strconv.Itoa(index)+"].path", value)
+		if err != nil {
+			return "", fmt.Errorf("failed to get kubelet config path: %w, couldn't find index of kc source", err)
+		}
+		fileContentStr := fileContent.(string)
+
+		// we found the path we're looking for
+		if fileContentStr == kcPath {
+			break
+		}
+		index++
+	}
+
+	kcPathFound := "storage.files[" + strconv.Itoa(index) + "].contents.source"
```

Comment:
If we iterate through the entire list of kubelet configs - this will be the last index + 1. Would that result in a runtime error, or do we have a test for that case?

Reply:
that's a good catch, just fixed that